jobs/kpi-forecasting/kpi_forecasting/tests/test_prophet_forecast.py

```diff
@@ -1,11 +1,544 @@
+    assert not pd.isna(output_df[non_metric_columns]).any(axis=None)
+
+
+def test_under_summarize(mocker, forecast):
+    """testing _summarize"""
+    # 2024-01-01 is chosen as an arbitrary date to center the tests around
+
+    # forecast predictions are set with the
+    # mock_aggregate_forecast_observed function so they
+    # can be ommited here
+    forecast_df = pd.DataFrame(
+        {
+            "submission_date": [
+                pd.to_datetime("2124-01-01").date(),
+                pd.to_datetime("2124-01-02").date(),
+            ],
+        }
+    )
+
+    # rows with negative values are those expected to be removed
+    # by filters in summarize
+    observed_df = pd.DataFrame(
+        {
+            "submission_date": [
+                pd.to_datetime("2124-01-01").date(),
+                pd.to_datetime("2124-01-02").date(),
+            ],
+            "value": [10, 20],
+        }
+    )
+
+    mocker.patch.object(
+        forecast, "_aggregate_forecast_observed", mock_aggregate_forecast_observed
+    )
+
+    numpy_aggregations = ["mean"]
+    percentiles = [10, 50, 90]
+    output_df = forecast._summarize(
+        forecast_df,
+        observed_df,
+        period="period",
+        numpy_aggregations=numpy_aggregations,
+        percentiles=percentiles,
+    )
+    observed_expected_df = pd.DataFrame(
+        {
+            "submission_date": [
+                pd.to_datetime("2124-01-01").date(),
+                pd.to_datetime("2124-01-02").date(),
+            ],
+            "value": [10, 20],
+            "measure": ["observed", "observed"],
+            "source": ["historical", "historical"],
+        }
+    )
+
+    # 4x2 columns, 4 metrics (mean, p10, p50, p90)
+    forecast_expected_df = pd.DataFrame(
+        {
+            "submission_date": [
+                pd.to_datetime("2124-01-01").date(),
+                pd.to_datetime("2124-01-02").date(),
+                pd.to_datetime("2124-01-01").date(),
+                pd.to_datetime("2124-01-02").date(),
+                pd.to_datetime("2124-01-01").date(),
+                pd.to_datetime("2124-01-02").date(),
+                pd.to_datetime("2124-01-01").date(),
+                pd.to_datetime("2124-01-02").date(),
+            ],
+            "measure": ["mean", "mean", "p10", "p10", "p50", "p50", "p90", "p90"],
+            "value": [0] * 8,
+            "source": ["forecast"] * 8,
+        }
+    )
+
+    # concat in same order to make our lives easier
+    expected = pd.concat([observed_expected_df, forecast_expected_df]).sort_values(
+        ["submission_date", "measure"]
+    )
+    expected["aggregation_period"] = "period"
+
+    assert set(expected.columns) == set(output_df.columns)
+    # force value columns to be floats in both cases to make check easier
+    numeric_cols = ["value", "value_low", "value_mid", "value_high"]
+    # expected[numeric_cols] = expected[numeric_cols].astype(float)
```

Comment:
Should these be commented out?

Reply:
I believe it's fine, for some reason in the funnel_forecast tests I was getting weird behavior with types so I just kind of threw my hands up and did that in the test.  Removed in latest commit.